scipy/stats/_continuous_distns.py

```diff
@@ -9577,10 +9579,11 @@ def solve_for_kappa(kappa):
         if floc is None:
             floc = find_mu(data)
             fshape = find_kappa(data)
-            return floc, fshape
         else:
             fshape = find_kappa(data)
-            return floc, fshape
+
+        floc = np.mod(floc + np.pi, 2 * np.pi) - np.pi  # ensure in [-pi, pi]
```

Comment:
In main, `loc` $\in [-\pi, \pi]$, but per [`vonmises` documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.vonmises.html):
![image](https://user-images.githubusercontent.com/6570539/219986415-9e9de2ad-e321-40f1-ab9f-9fad9b64c812.png)

Also, previously:
```python3
import numpy as np
from scipy import stats
rng = np.random.default_rng(356983576)

size =  10000
shape, loc, scale = 100, -0.5, 1
dist = stats.vonmises(shape, loc=loc, scale=scale)
rvs = dist.rvs(size=size, random_state=rng)
stats.vonmises.fit(rvs, fscale=1, superfit=True)  # (100.41820385099112, -0.4987617235978734, 1)
# location is -0.5
```